pkg/manager/cli_download_install.go

```diff
@@ -0,0 +1,319 @@
+	routev1 "github.com/openshift/api/route/v1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func EnableHypershiftCLIDownload(hubclient client.Client, log logr.Logger) error {
+	// get the current version of MCE CSV from multicluster-engine namespace
+	csv, err := GetMCECSV(hubclient, log)
+	if err != nil {
+		log.Error(err, "failed to get the most current version of MCE CSV from multicluster-engine namespace")
+		return err
+	}
+
+	// check if the CSV has hypershift_cli image, which is the downstream case
+	cliDownloadImage := GetHypershiftCLIDownloadImage(csv, log)
+	if cliDownloadImage == "" {
+		// in upstream build, there is no hypershift CLI download image
+		log.Info("the hypershift CLI download image was not found in the CSV. Skip enabling the hypershift CLI download")
+		return nil
+	}
+
+	err = DeployHypershiftCLIDownload(hubclient, cliDownloadImage, log)
+	if err != nil {
+		log.Error(err, "failed to deploy HypershiftCLIDownload")
+		return err
+	}
+
+	return err
+}
+
+func GetMCECSV(hubclient client.Client, log logr.Logger) (*operatorsv1alpha1.ClusterServiceVersion, error) {
+	csvlist := &operatorsv1alpha1.ClusterServiceVersionList{}
+
+	listopts := &client.ListOptions{Namespace: "multicluster-engine"}
+
+	err := hubclient.List(context.TODO(), csvlist, listopts)
+
+	if err != nil {
+		log.Error(err, "failed to list CSVs")
+		return nil, err
+	}
+
+	var names []string
+	for _, csv := range csvlist.Items {
+		if strings.HasPrefix(csv.Name, "multicluster-engine.") {
+			names = append(names, csv.Name)
+		}
+	}
+
+	if len(names) == 0 {
+		err := errors.New("no MCE CSV found")
+		log.Error(err, "no MCE CSV found")
+		return nil, err
+	}
+
+	// find the latest version of MCE CSV
+	sort.Sort(sort.Reverse(sort.StringSlice(names)))
+
+	csv := &operatorsv1alpha1.ClusterServiceVersion{}
+	csvNN := types.NamespacedName{Namespace: "multicluster-engine", Name: names[0]}
+
+	err = hubclient.Get(context.TODO(), csvNN, csv)
+
+	if err != nil {
+		log.Error(err, "failed to get MCE CSV "+names[0])
+		return nil, err
+	}
+
+	log.Info("MCE CSV found " + csv.Name)
+	return csv, nil
+}
+
+func GetHypershiftCLIDownloadImage(csv *operatorsv1alpha1.ClusterServiceVersion, log logr.Logger) string {
+	for _, relatedImage := range csv.Spec.RelatedImages {
+		if strings.EqualFold(relatedImage.Name, "hypershift_cli") && relatedImage.Image != "" {
+			log.Info("the hypershift CLI download image was found in the CSV " + relatedImage.Image)
+			return relatedImage.Image
+		}
+	}
+
+	return ""
+}
+
+func DeployHypershiftCLIDownload(hubclient client.Client, cliImage string, log logr.Logger) error {
```

Comment:
Should this func name be lowercase deployHypershiftCLIDownload

Reply:
Yes